src/bin/main.rs

```diff
@@ -25,16 +25,35 @@ use std::path::PathBuf;
 type Pool = r2d2::Pool<ConnectionManager<PgConnection>>;
 
 fn db_pool(rocket: &Rocket) -> Pool {
-    // TODO: Convert this to build the database_url using values from the app config.
-    let database_url = rocket.config().get_str("database_url").expect("Must set DATABASE_URL");
+    let database_url = rocket.config().get_str("database_url").expect("Must set Database.url in config");
     let manager = ConnectionManager::<PgConnection>::new(database_url);
     r2d2::Pool::builder().build(manager).expect("Could not get DB connection pool")
 }
 
 fn app(config: config::Config) -> Rocket {
+    let mut db_url = String::new();
+    match config.get_str("Database.type").unwrap().to_lowercase().as_str() {
+        "postgresql" => db_url.push_str("postgres://"),
```

Comment:
Since postgres connection strings accept `postgres://` and `postgresql://`, we should probably allow both in the config file too.

Reply:
I'm concerned that may lead to some confusion about possible values for for Database.type in config.toml. Then again, it could also make that value a little more robust when we eventually add support for other DBMSs like MySQL. As long as there is no difference between "postgres://" and "postgresql://" in the connection string, I'm fine with either way.

Given these thoughts, which way do you prefer?

Reply:
I'm still fine with either. If there turns out to be confusion with what can be put there, it is easy enough to add "this config variable takes X, Y, and Z as possible values" to the docs.

Reply:
Implemented by c2b14a3